app.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'sinatra/reloader'
+require 'csv'
+
+set :strict_paths, false
+
+before do
+  data_dir = File.join(Dir.pwd, 'data')
+  @csv_path = File.join(data_dir, 'memos.csv')
+  @max_id_path = File.join(data_dir, 'max_id.txt')
+
+  Dir.mkdir(data_dir) unless Dir.exist?(data_dir)
+  prepare_max_id_file(@max_id_path)
+  prepare_data_file(@csv_path)
+end
+
+get '/memos' do
+  @page_title = 'メモ一覧'
+  @memos = []
+
+  CSV.foreach(@csv_path, headers: true) do |memo|
+    @memos << { id: memo['id'], title: memo['title'] }
+  end
+  erb :memos
+end
+
+get '/' do
+  redirect '/memos'
+end
+
+get '/memos/new' do
+  @page_title = '新規作成'
+  erb :new
+end
+
+post '/memos' do
+  title = protect_xss(params[:title])
+  content = protect_xss(params[:content])
+  id = File.read(@max_id_path).to_i + 1
+
+  CSV.open(@csv_path, 'a', quote_char: '"') { |csv| csv << [id, title, content] }
+  File.open(@max_id_path, 'w') { |file| file << id }
+  redirect "/memos/#{id}"
+end
```

Comment:
引数の使われ方や意味のまとまり的に以下のようにするのがよいかなと思います。

```suggestion
post '/memos' do
  # メモの準備
  id = File.read(@max_id_path).to_i + 1
  title = protect_xss(params[:title])
  content = protect_xss(params[:content])

  # メモの保存
  CSV.open(@csv_path, 'a', quote_char: '"') { |csv| csv << [id, title, content] }
  File.open(@max_id_path, 'w') { |file| file << id }

  # 画面の遷移
  redirect "/memos/#{id}"
end
```

Reply:
そのように修正します。